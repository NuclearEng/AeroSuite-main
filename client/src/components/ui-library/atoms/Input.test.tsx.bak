import React from 'react';
import renderer from 'react-test-renderer';
import Input from './Input';
import { ThemeProvider } from '@mui/material/styles';
import { createTheme } from '@mui/material/styles';

// Create a test theme to ensure consistent snapshots
const testTheme = createTheme({
  palette: {
    primary: {
      main: '#1976d2',
    },
    secondary: {
      main: '#dc004e',
    },
  },
});

describe('Input Component', () => {
  it('renders correctly with default props', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={testTheme}>
          <Input placeholder="Enter text" />
        </ThemeProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with label', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={testTheme}>
          <Input label="Username" placeholder="Enter username" />
        </ThemeProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with error state', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={testTheme}>
          <Input 
            error 
            helperText="This field is required" 
            label="Email" 
            placeholder="Enter email" 
          />
        </ThemeProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly when disabled', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={testTheme}>
          <Input 
            disabled 
            label="Disabled Field" 
            placeholder="Cannot edit this field" 
          />
        </ThemeProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with different variants', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={testTheme}>
          <div>
            <Input variant="outlined" label="Outlined" />
            <Input variant="filled" label="Filled" />
            <Input variant="standard" label="Standard" />
          </div>
        </ThemeProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
}); 