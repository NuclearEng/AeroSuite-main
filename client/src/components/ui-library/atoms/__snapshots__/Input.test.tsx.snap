// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component renders correctly when disabled 1`] = `null`;

exports[`Input Component renders correctly with default props 1`] = `null`;

exports[`Input Component renders correctly with different variants 1`] = `null`;

exports[`Input Component renders correctly with error state 1`] = `null`;

exports[`Input Component renders correctly with label 1`] = `null`;
